src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java

```diff
@@ -209,6 +212,20 @@
     public static final Permission MANAGE_DOMAINS = new Permission(GROUP, "ManageDomains",
             Messages._CredentialsProvider_ManageDomainsPermissionDescription(), Permission.CONFIGURE, true, SCOPES);
 
+    /**
+     * The system property name corresponding to {@link #FINGERPRINT_ENABLED}.
+     */
+    public static final String FINGERPRINT_ENABLED_NAME = CredentialsProvider.class.getSimpleName() + ".fingerprintEnabled";
+    
+    /**
+     * Control if the fingerprints must be used or not. 
+     * By default they are activated and thus allow the tracking of credentials usage.
+     * In case of performance troubles in some weird situation, you can disable the behavior by setting it to {@code false}.
+     */
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "Accessible via System Groovy Scripts")
+    @Restricted(NoExternalUse.class)
+    public static /* not final */ boolean FINGERPRINT_ENABLED = SystemProperties.getBoolean(FINGERPRINT_ENABLED_NAME, true);
```

Comment:
There is some discussion elsewhere on whether this form should be used.

Reply:
Please provide a link to the related discussion

Reply:
So far the only counter-argument I've seen is use from `jenkins-test-harness`.

Reply:
See [discussion](https://github.com/jenkinsci/jenkins/pull/4389#discussion_r369716605).